### PR TITLE
cmd/geth: fix init genesis for dev

### LINKED ISSUE #24453: Geth does not start in --dev mod with empty blockchain and custom genesis block
#### System information

Geth version: `1.10.16`
OS & Version: Ubuntu 21.10 Implish

#### Expected behaviour

If run
```bash
geth --datadir /tmp/blockchain/ init genesis.json
geth --datadir /tmp/blockchain --dev
```

geth starts in develop mode.

#### Actual behaviour

geth fails with an error:
```text
Fatal: Could not open database: open /tmp/blockchain/geth/chaindata/ancient/headers.cidx: no such file or directory
```

#### Steps to reproduce the behaviour

1. Create valid genesis block and write in to `genesis.json` file
2. Run `geth --datadir /tmp/blockchain/ init genesis.json`
3. Run `geth --datadir /tmp/blockchain --dev`

#### Backtrace

````
INFO [02-22|17:03:03.930] Starting Geth in ephemeral dev mode... 
INFO [02-22|17:03:03.931] Maximum peer count                       ETH=50 LES=0 total=50
INFO [02-22|17:03:03.931] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [02-22|17:03:03.931] Set global gas cap                       cap=50,000,000
INFO [02-22|17:03:04.078] Using developer account                  address=0x426b4b03F4d92Fc4c45dC9858B9D62BC9625C853
INFO [02-22|17:03:04.078] Allocated cache and file handles         database=/tmp/blockchain/geth/chaindata cache=512.00MiB handles=524,288 readonly=true
Fatal: Could not open database: open /tmp/blockchain/geth/chaindata/ancient/headers.cidx: no such file or directory
````

### PR DESCRIPTION
**Pull request by s1na**
Fixes #24453

### CODE REVIEW

File: cmd/geth/chaincmd.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func initGenesis(ctx *cli.Context) error {
 	defer stack.Close()
 
 	for _, name := range []string{"chaindata", "lightchaindata"} {
-		chaindb, err := stack.OpenDatabase(name, 0, 0, "", false)
```
Comment (rjl493456442) [inline cmd/geth/chaincmd.go:209]: In this case, the freezer is always initialized with the default path `<DATADIR>/chaindata/ancient`.
Although we never offer the ancient flag for customizing the ancient path though(it should be fixed).

Comment (s1na) [reply cmd/geth/chaincmd.go:209]: Yep, I'll wait for https://github.com/ethereum/go-ethereum/pull/24698 to get merged and take ancient path from the flag added there

Comment (holiman) [reply cmd/geth/chaincmd.go:209]: Ping, #24698 merged

Comment (s1na) [reply cmd/geth/chaincmd.go:196]: Updated

**Review by holiman — APPROVED** [review]

**Pull request merged**